x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/http/sender/HttpRequestSenderFactoryTests.java

```diff
@@ -0,0 +1,119 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.sender;
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.http.MockResponse;
+import org.elasticsearch.test.http.MockWebServer;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.inference.external.http.HttpClientManager;
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.xpack.inference.external.http.HttpClientManagerTests.mockClusterServiceEmpty;
+import static org.elasticsearch.xpack.inference.external.http.HttpClientTests.createHttpPost;
+import static org.elasticsearch.xpack.inference.external.http.HttpClientTests.createThreadPool;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HttpRequestSenderFactoryTests extends ESTestCase {
+    private static final TimeValue TIMEOUT = new TimeValue(30, TimeUnit.SECONDS);
+    private final MockWebServer webServer = new MockWebServer();
+    private ThreadPool threadPool;
+    private HttpClientManager clientManager;
+    private Thread thread;
+
+    @Before
+    public void init() throws Exception {
+        webServer.start();
+        threadPool = createThreadPool(getTestName());
+        clientManager = HttpClientManager.create(Settings.EMPTY, threadPool, mockClusterServiceEmpty());
+        thread = null;
+    }
+
+    @After
+    public void shutdown() throws IOException, InterruptedException {
+        if (thread != null) {
+            // thread.interrupt();
+            thread.join(TIMEOUT.millis());
+        }
+
+        clientManager.close();
+        terminate(threadPool);
+        webServer.close();
+    }
+
+    public void testCreateSender_SendsRequestAndReceivesResponse() throws Exception {
+        var mockExecutorService = mock(ExecutorService.class);
+        doAnswer(invocation -> {
+            Runnable runnable = (Runnable) invocation.getArguments()[0];
+            thread = new Thread(runnable);
+            thread.start();
```

Comment:
This is gross but for some reason this test was waiting 10 seconds when trying to start the queuing request executor. So using a thread directly avoids that.

Reply:
I wonder what the cause of the 10 second wait is 🤔

Reply:
From my testing it had something to do with the fact that the thread is getting blocked in the `queue.take()`. I figured that'd be fine an it'd just spin up another thread, which does seem to eventually happen but it takes a while for some reason.